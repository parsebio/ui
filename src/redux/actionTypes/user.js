const USER = 'user';

/**
 * User was loaded
 */
const USER_LOADED = `${USER}/loaded`;

/**
 * User was not authenticated
 */
const USER_NOT_AUTHENTICATED = `${USER}/notAuthenticated`;

export {
  // eslint-disable-next-line import/prefer-default-export
  USER_LOADED,
  USER_NOT_AUTHENTICATED,
};
