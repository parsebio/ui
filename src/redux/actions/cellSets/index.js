import createCellSet from './createCellSet';
import deleteCellSet from './deleteCellSet';
import deleteCellClass from './deleteCellClass';
import runCellSetsClustering from './runCellSetsClustering';
import runCellSetsAnnotation from './runCellSetsAnnotation';
import updateCellSetsClustering from './updateCellSetsClustering';

import loadCellSets from './loadCellSets';
import reorderCellSet from './reorderCellSet';
import updateCellSetProperty from './updateCellSetProperty';
import setCellSetHiddenStatus from './setCellSetHiddenStatus';
import unhideAllCellSets from './unhideAllCellSets';

export {
  createCellSet,
  deleteCellSet,
  deleteCellClass,
  runCellSetsClustering,
  runCellSetsAnnotation,
  updateCellSetsClustering,
  loadCellSets,
  reorderCellSet,
  updateCellSetProperty,
  setCellSetHiddenStatus,
  unhideAllCellSets,
};
