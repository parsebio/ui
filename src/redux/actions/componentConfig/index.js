import loadPlotConfig from 'redux/actions/componentConfig/loadPlotConfig';
import loadComponentConfig from 'redux/actions/componentConfig/loadComponentConfig';
import updatePlotConfig from 'redux/actions/componentConfig/updatePlotConfig';
import replaceLoadedConfigs from 'redux/actions/componentConfig/replaceLoadedConfigs';
import savePlotConfig from 'redux/actions/componentConfig/savePlotConfig';
import updatePlotData from 'redux/actions/componentConfig/updatePlotData';
import resetPlotConfig from 'redux/actions/componentConfig/resetPlotConfig';
import getDotPlot from 'redux/actions/componentConfig/getDotPlot';
import downloadNormalizedMatrix from 'redux/actions/componentConfig/downloadNormalizedMatrix';
import clearPlotData from 'redux/actions/componentConfig/clearPlotData';

export {
  loadPlotConfig,
  loadComponentConfig,
  updatePlotConfig,
  replaceLoadedConfigs,
  savePlotConfig,
  updatePlotData,
  resetPlotConfig,
  getDotPlot,
  downloadNormalizedMatrix,
  clearPlotData,
};
