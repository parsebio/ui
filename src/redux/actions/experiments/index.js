import createExperiment from './createExperiment';
import deleteExperiment from './deleteExperiment';
import updateExperiment from './updateExperiment';
import loadExperiments from './loadExperiments';
import loadExampleExperiments from './loadExampleExperiments';
import setActiveExperiment from './setActiveExperiment';
import switchExperiment from './switchExperiment';
import reorderSamples from './reorderSamples';
import createMetadataTrack from './createMetadataTrack';
import renameMetadataTrack from './renameMetadataTrack';
import deleteMetadataTrack from './deleteMetadataTrack';
import updateValueInMetadataTrack from './updateValueInMetadataTrack';
import cloneExperiment from './cloneExperiment';
import reloadExperimentInfo from './reloadExperimentInfo';
import updateCellLevelMetadataFileUpload from './updateCellLevelMetadataFileUpload';
import createCellLevelMetadata from './createCellLevelMetadata';
import deleteCellLevelMetadata from './deleteCellLevelMetadata';

export {
  createExperiment,
  deleteExperiment,
  updateExperiment,
  loadExperiments,
  loadExampleExperiments,
  setActiveExperiment,
  switchExperiment,
  reorderSamples,
  createMetadataTrack,
  renameMetadataTrack,
  deleteMetadataTrack,
  updateValueInMetadataTrack,
  cloneExperiment,
  reloadExperimentInfo,
  updateCellLevelMetadataFileUpload,
  createCellLevelMetadata,
  deleteCellLevelMetadata,
};
