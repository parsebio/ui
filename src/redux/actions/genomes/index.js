import createAndUploadGenomeFile from 'redux/actions/genomes/createAndUploadGenomeFile';
import updateGenomeFileUploadProgress from 'redux/actions/genomes/updateGenomeFileUploadProgress';
import createGenome from 'redux/actions/genomes/createGenome';
import loadGenomes from 'redux/actions/genomes/loadGenomes';

export {
  createGenome,
  createAndUploadGenomeFile,
  updateGenomeFileUploadProgress,
  loadGenomes,
};
