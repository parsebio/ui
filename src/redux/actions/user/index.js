import loadUser from './loadUser';

export {
  // eslint-disable-next-line import/prefer-default-export
  loadUser,
};
