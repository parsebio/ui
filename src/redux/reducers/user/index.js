import {
  USER_LOADED,
} from 'redux/actionTypes/user';

import initialState from 'redux/reducers/user/initialState';
import userLoaded from 'redux/reducers/user/userLoaded';

const userReducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case USER_LOADED: {
      return userLoaded(state, action);
    }

    default: {
      return state;
    }
  }
};

export default userReducer;
