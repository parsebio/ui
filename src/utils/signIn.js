import { Auth } from 'aws-amplify';
import Cookies from 'js-cookie';

import cache from 'utils/cache';

const signIn = async () => {
  // Clear old cookies that Amplify might have left behind.
  // This avoids infinite redirect loops when the user tries to sign in
  Object.keys(Cookies.get())
    .filter((key) => key.startsWith('CognitoIdentityServiceProvider.'))
    .forEach((key) => Cookies.remove(key));

  // Store current url so that post-login we can send them to the place they expected
  cache.set('redirectUrl', window.location.href);
  await Auth.federatedSignIn();
};

export default signIn;
