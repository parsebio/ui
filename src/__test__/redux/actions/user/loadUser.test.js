import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import fetchMock from 'jest-fetch-mock';
import { loadUser } from 'redux/actions/user';
import { USER_LOADED } from 'redux/actionTypes/user';
import signIn from 'utils/signIn';
import { Auth } from '@aws-amplify/auth';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

jest.mock('utils/signIn');
jest.mock('aws-amplify', () => ({
  Auth: {
    currentAuthenticatedUser: jest.fn().mockImplementation(async () => ({
      attributes: {
        'custom:agreed_terms_v2': 'true',
        'custom:agreed_cookies_v1': 'true',
        name: 'Tester Testson',
      },
    })),
    federatedSignIn: jest.fn(),
  },
}));

describe('loadUser action', () => {
  let store;

  beforeEach(() => {
    fetchMock.enableMocks();
    store = mockStore({});
  });

  afterEach(() => {
    fetchMock.disableMocks();
    fetchMock.resetMocks();
  });

  it('dispatches USER_LOADED when user is successfully loaded', async () => {
    fetchMock.mockResponseOnce(JSON.stringify({ user: { name: 'Test User' } }));

    await store.dispatch(loadUser());

    const actions = store.getActions();
    expect(actions[0].type).toEqual(USER_LOADED);
    expect(actions[0].payload).toMatchSnapshot();
  });

  it('handles exceptions by not dispatching USER_LOADED', async () => {
    Auth.currentAuthenticatedUser = jest.fn().mockImplementationOnce(async () => {
      throw new Error('Failed to fetch');
    });

    await store.dispatch(loadUser());

    const actions = store.getActions();
    expect(signIn).toHaveBeenCalled();
    expect(actions).toEqual([]);
  });
});
