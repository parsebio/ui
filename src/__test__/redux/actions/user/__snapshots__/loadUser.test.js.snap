// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadUser action dispatches USER_LOADED when user is successfully loaded 1`] = `
{
  "user": {
    "attributes": {
      "custom:agreed_cookies_v1": "true",
      "custom:agreed_terms_v2": "true",
      "name": "Tester Testson",
    },
  },
}
`;
